### ISSUE TITLE
# Sample State:

### ISSUE THREAD

**Issue by TahjDev** [EthanLam17/artify#3]
- [x] State shape is flat!
- [x] State's keys are camelCased
- [x] All keys within the values in the state are accessible in the schema
- [x] Correctly formatted
    - [x] Sample state is rendered with triple backticks, and the language ```javascript...```). This will display the state as a code block instead of a giant line of text
    - [x] Top level slices
        - [x] entities
        - [x] session
        - [x] errors (here or in ui)
        - [x] ui (if needed)
    - [x] Should NOT have nested slices, aka comments inside of posts
        - Some info from other tables is ok, for instance:
            - the author username and imageurl for a post. basically any info that the user can't change
            - like count and a boolean on whether the user likes the post instead of a likes slice

**Comment by TahjDev** [comment]
- Why does artistId point to an array? is this for the main artist and their features? and if so then you not a table to join songs to artists and could the name be artistIds plural just to be a bit more intuitive
- I don't see a UI slice of state,
- How are playlists and songs connected cause a playlist can have many songs and vice verse

**Comment by EthanLam17** [comment]
Yes you are correct, artist Id points to array because I thought that would be the easiest way to show the multiple artists that could be associated with a song. Do you think it is recommended to switch to a joins table? I am slightly reluctant to because I feel like it still works with my way of using an array to hold the associated artists to a song. 

I do not think I have any UI slices of state. It seems like people usually use it for modals which I do not plan to include in this project.

**Comment by TahjDev** [comment]
- So you need a joins table for songs and artists because if a song can have many artists and artists can have many songs how would you represent that relationship? If you have just a song table with an artist_id, a song can only belong to a single artist.

**Comment by EthanLam17** [comment]
Thanks Tahj! That makes perfect sense, honestly I haven't been thinking about the full stack much since I've been focused on the exam. I have update the sample state and hopefully it is good to do. 

Side note: You may notice that the first song has an album id of 0. I did this to indicate that all albums with an ID of 0 are actually singles and are not associated with an album. Is this a viable way of doing this or will there be issues with it also being falsey. Alternatively I can make it an arbitrary number like 999.
